keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -620,22 +625,83 @@ class SqlArtifactRepository(
              * or other issue unrelated to an artifact version triggering continual automated rollbacks
              * thru all previously deployed versions.
              */
-            if (prior != null) {
-              txn.update(ENVIRONMENT_ARTIFACT_VERSIONS)
-                .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE, veto.version)
-                .where(
-                  ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid),
-                  ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artUid),
-                  ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(prior)
-                )
-                .execute()
-            }
+            prior?.let { txn.setPromotionReference(veto.version, envUid, artUid, it) }
           }
         }
         return@fetchOne true
       }
   }
 
+  private fun DSLContext.setPromotionReference(version: String, envUid: String, artUid: String, prior: String) {
+    update(ENVIRONMENT_ARTIFACT_VERSIONS)
+      .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE, version)
+      .where(
+        ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid),
+        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artUid),
+        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(prior)
+      )
+      .execute()
+  }
+
+  private fun DSLContext.addRecordToEnvironmentArtifactVetoTable(envUid: String, artUid: String, veto: EnvironmentArtifactVeto) {
+    insertInto(ENVIRONMENT_ARTIFACT_VETO)
+      .set(ENVIRONMENT_ARTIFACT_VETO.ENVIRONMENT_UID, envUid)
+      .set(ENVIRONMENT_ARTIFACT_VETO.ARTIFACT_UID, artUid)
+      .set(ENVIRONMENT_ARTIFACT_VETO.ARTIFACT_VERSION, veto.version)
+      .set(ENVIRONMENT_ARTIFACT_VETO.VETOED_AT, clock.timestamp())
+      .set(ENVIRONMENT_ARTIFACT_VETO.VETOED_BY, veto.vetoedBy)
+      .set(ENVIRONMENT_ARTIFACT_VETO.COMMENT, veto.comment)
+      .onDuplicateKeyUpdate()
+      .set(ENVIRONMENT_ARTIFACT_VETO.VETOED_AT, clock.timestamp())
+      .set(ENVIRONMENT_ARTIFACT_VETO.VETOED_BY, veto.vetoedBy)
+      .set(ENVIRONMENT_ARTIFACT_VETO.COMMENT, veto.comment)
+      .execute()
+  }
+
+  private fun DSLContext.updateAsVetoedInEnvironmentArtifactVersionsTable(prior: String?, veto: EnvironmentArtifactVeto, envUid: String, artUid: String) {
+    update(ENVIRONMENT_ARTIFACT_VERSIONS)
+      .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS, VETOED.name)
+      .set(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE, prior ?: veto.version)
+      .where(
+        ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid),
+        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artUid),
+        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(veto.version)
+      )
+      .execute()
+  }
+
+  private fun selectArtifactStatusAndPromotionReferenceInTargetEnvironment(envUid: String, artUid: String, version: String): SelectConditionStep<Record2<String, String>> {
```

Comment:
nit: bump the vars and return type down on other lines so this isn't so long

Reply:
fixed in d6d12b1